pkg/tsdb/grafanads/query.go

```diff
@@ -7,6 +7,9 @@ const (
 	// QueryTypeList will list the files in a folder
 	queryTypeSearch = "search"
 
+	// QueryTypeSearchReadiness will indicate whether search is ready to serve requests
+	queryTypeSearchReadiness = "searchReadiness"
```

Comment:
rather than this... could we have the search return 202, and then have the frontend try again with SQL flavor?

Reply:
I don't mind, but I am not sure how would that work, as we are using the [DS API](https://github.com/grafana/grafana/blob/803473f479fe872129b042704c4b9cdad91ed849/packages/grafana-runtime/src/utils/DataSourceWithBackend.ts#L112) for search rather than our own, custom one. Are you suggesting some custom [LoadingState](https://github.com/grafana/grafana/blob/9db9ebce0254b0202cdf29ec3355fbd4e8c22f73/packages/grafana-data/src/types/datasource.ts#L446) / dataframe format?

Reply:
Also, I assume we will still need to somehow handle this edge case once we remove the SQL API (eg. with info on the UI). Wouldn't that extra query type be useful for that scenario? Or do you think we are going to get rid of this issue altogether?